app/views/users/show.html.slim

```diff
@@ -36,38 +40,39 @@ javascript:
               span.fa.fa-edit
               |  Edit Profile
       
-      - unless @ban.nil?
+      - unless @ban.nil? || @user.marked_for_death
         .card class="alert alert-danger" style="padding: 0px;"
           .card-body style="text-align: center;" 
             b Banned until #{date_string_short(@ban.expiration)}
             br
             | Ban Reason: #{@ban.reason}
-            - if logged_in_as_moderator
-              br
-              = link_to "Lift Ban", "#{user_path(@user)}/lift_ban", method: :post, class:"btn btn-primary", style: "margin-left: 5px; margin-top: 5px;"
 
-      - if logged_in_as_moderator
+      - if logged_in_as_moderator && !@user.marked_for_death
```

Comment:
Add `@user.banned?`. Also, I missed this before, but please stop setting redundant `@`variables. Set up an association in the model and just use `@user.ban`.